services/web/server/src/simcore_service_webserver/storage_handlers.py

```diff
@@ -118,3 +128,76 @@ async def delete_file(request: web.Request):
     await check_permission(request, "storage.files.*")
     payload = await _request_storage(request, "DELETE")
     return payload
+
+
+async def get_project_files_metadata(
+    app: web.Application, location_id: str, uuid_filter: str, user_id: int
+) -> List[Dict[str, Any]]:
+    session = get_client_session(app)
+
+    url: URL = (
+        _get_base_storage_url(app) / "locations" / location_id / "files" / "metadata"
+    )
+    params = dict(user_id=user_id, uuid_filter=uuid_filter)
+    async with session.get(url, ssl=False, params=params) as resp:
+        payload = await resp.json()
+        if not isinstance(payload, dict):
+            raise web.HTTPException(reason=f"Did not receive a dict: '{payload}'")
+        if "data" not in payload:
+            raise web.HTTPException(reason=f"No url found in response: '{payload}'")
+        if not isinstance(payload["data"], list):
+            raise web.HTTPException(
+                reason=f"No list payload received as data: '{payload}'"
+            )
+
+        return payload["data"]
+
+
+async def get_file_download_url(
+    app: web.Application, location_id: str, fileId: str, user_id: int
+) -> str:
+    session = get_client_session(app)
+
+    url: URL = (
+        _get_base_storage_url(app)
+        / "locations"
+        / location_id
+        / "files"
+        / urllib.parse.quote(fileId, safe="")
+    )
+    params = dict(user_id=user_id)
+    async with session.get(url, ssl=False, params=params) as resp:
+        payload = await resp.json()
+        if not isinstance(payload, dict):
+            raise web.HTTPException(reason=f"Did not receive a dict: '{payload}'")
+        if "data" not in payload:
+            raise web.HTTPException(reason=f"No url found in response: '{payload}'")
+        if "link" not in payload["data"]:
+            raise web.HTTPException(reason=f"No url found in response: '{payload}'")
+
+        return payload["data"]["link"]
+
+
+async def get_file_upload_url(
+    app: web.Application, location_id: str, fileId: str, user_id: int
+) -> str:
+    session = get_client_session(app)
+
+    url: URL = (
+        _get_base_storage_url(app)
+        / "locations"
+        / location_id
+        / "files"
+        / urllib.parse.quote(fileId, safe="")
+    )
+    params = dict(user_id=user_id)
+    async with session.put(url, ssl=False, params=params) as resp:
+        payload = await resp.json()
+        if not isinstance(payload, dict):
+            raise web.HTTPException(reason=f"Did not receive a dict: '{payload}'")
```

Comment:
and here

Reply:
I am aware, waiting for the PR to merge it

Reply:
no I'm talking about unwraping the response